packages/@lwc/ssr-runtime/src/index.ts

```diff
@@ -172,20 +172,32 @@ export class LightningElement implements PropsAvailableAtConstruction {
         return (this.__classList = new ClassList(this));
     }
 
-    getAttribute(attrName: string): string | null {
-        return this.__attrs[attrName] ?? null;
+    #setAttribute(attrName: string, attrValue: string | null): void {
+        this.__attrs[attrName] = attrValue;
     }
 
-    setAttribute(attrName: string, value: string | null): void {
-        this.__attrs[attrName] = String(value);
+    setAttribute(attrName: string, attrValue: unknown): void {
+        this.#setAttribute(attrName, String(attrValue));
     }
 
-    hasAttribute(attrName: string): boolean {
-        return Boolean(this.__attrs && attrName in this.__attrs);
+    getAttribute(attrName: unknown): string | null {
+        if (this.hasAttribute(attrName)) {
+            return this.__attrs[attrName as string];
+        }
+        return null;
+    }
+
+    hasAttribute(attrName: unknown): boolean {
+        return typeof attrName === 'string' && typeof this.__attrs[attrName] === 'string';
```

Comment:
The fact that we have to check for strings here because we don't ensure that the values in `this.__attrs` is only a string type is odd to me. We should probably do the validation/coercion on the way in.

Reply:
OK, so the reason for this is that we can't call `delete` because that would delete the getter/setter which is required for prop/attr reflection.

Reply:
https://github.com/salesforce/lwc/issues/4641